File: LQRefresh/UIScrollView+LQRefresh.m
//
//  UIScrollView+LQRefresh.m
//  LQRefreshDemo
//
//  Created by hongzhiqiang on 2018/11/13.
//  Copyright © 2018 hhh. All rights reserved.
//

#import "UIScrollView+LQRefresh.h"
#import "LQRefreshHeader.h"
#import <objc/runtime.h>
@implementation UIScrollView (LQRefresh)


- (void)setLq_header:(LQRefreshHeader *)lq_header {
    if (lq_header != self.lq_header) {
        [self.lq_header removeFromSuperview];
        [self insertSubview:lq_header atIndex:0];
        objc_setAssociatedObject(self, "lq_header", lq_header, OBJC_ASSOCIATION_RETAIN);
    }
}

- (LQRefreshHeader *)lq_header {
   return objc_getAssociatedObject(self, "lq_header");
}


@end

File: LQRefreshSwift/UIScrollView+LQRefresh.swift
//
//  UIScrollView+LQRefresh.swift
//  LQRefreshSwiftDemo
//
//  Created by hhh on 2018/11/28.
//  Copyright © 2018 LQ. All rights reserved.
//

import UIKit

var LQRefreshHeaderKey = 110

extension UIScrollView {
    var lq_header: LQRefreshHeader? {
        set (newValue) {
            if newValue != nil {
                self.lq_header?.removeFromSuperview()
                self.insertSubview(newValue!, at: 0)
                objc_setAssociatedObject(self, &LQRefreshHeaderKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN)
            }
        }
        get {
            return objc_getAssociatedObject(self, &LQRefreshHeaderKey) as? LQRefreshHeader
        }
    }
    
}
